File: php/conjecture.php
<?php
# Collatz conjecture
# Seeming randomness in mathematics
# aka  : 3x+1
# video: https://www.youtube.com/watch?v=094y1Z2wpJg

$num = rand(0, 10000);
$startnum = $num;
$count = 0;

printf("\n=== Collatz conjecture. Using %d ===\n\n", $num); 

while(True) {
  if($num % 2 == 0) {
    $num = $num / 2;
    printf("\033[91m%d\033[0m\n", $num);
  } else {
    $num = ($num * 3) + 1;
    printf("\033[1m%d\033[0m\n", $num);
  }

  if($num == 1) {
    printf("\nThe number %d reached 1 in %d steps, stopping to not loop forever.\n", $startnum, $count);
    break;
  }
  $count++;
}

?>

File: java/conjecture.java
// Collatz conjecture
// Seeming randomness in mathematics
// aka  : 3x+1
// video: https://www.youtube.com/watch?v=094y1Z2wpJg

import java.util.Random;

class conjecture {
  public static void main(String[] args) {
    Random rand = new Random();
    int num = rand.nextInt(10000);
    
    int startnum = num;
    int count = 0;
    
    String str = String.format("\n=== Collatz conjecture. Using %d ===", startnum);
    System.out.println(str); 

    while(true) {
      if(num % 2 == 0) {
        num = num / 2;
        System.out.println(String.format("\033[91m%d\033[0m", num));
      } else {
        num = (num * 3) + 1;
        System.out.println(String.format("\033[1m%d\033[0m", num));
      }
      
      count++;

      if(num == 1) {
        break;
      }
    }
    System.out.println(String.format("\nThe number %d reached 1 in %d steps, stopping to not loop forever.\n", count, num));
  }
}
